lib/src/common/disposable.dart

```diff
@@ -351,6 +352,32 @@ class Disposable implements _Disposable, DisposableManagerV3, LeakFlagger {
 
   @mustCallSuper
   @override
+  StreamSubscription<T> getManagedStreamSubscription<T>(
+      Stream<T> stream, void onData(T event)) {
+    _throwOnInvalidCall2('awaitBeforeDispose', 'future', stream, onData);
```

Comment:
I've just made two calls to `_throwOnInvalidCall` in the past. There is a tiny bit of repetition, but not much. This is fine, though.